salt/utils/win_functions.py

```diff
@@ -210,3 +212,72 @@ def escape_meta_chars(m):
         return meta_map[char]
 
     return meta_re.sub(escape_meta_chars, arg)
+
+
+def broadcast_setting_change(message='Environment'):
+    '''
+    Send a WM_SETTINGCHANGE Broadcast to all Windows
+
+    Args:
+
+        message (str):
+            A string value representing the portion of the system that has been
+            updated and needs to be refreshed. Default is ``Environment``. These
+            are some common values:
+
+            - "Environment" : to effect a change in the environment variables
+            - "intl" : to effect a change in locale settings
+            - "Policy" : to effect a change in Group Policy Settings
+            - a leaf node in the registry
+            - the name of a section in the ``Win.ini`` file
+
+            See lParam within msdn docs for
+            `WM_SETTINGCHANGE <https://msdn.microsoft.com/en-us/library/ms725497%28VS.85%29.aspx>`
+            for more information on Broadcasting Messages.
+
+            See GWL_WNDPROC within msdn docs for
+            `SetWindowLong <https://msdn.microsoft.com/en-us/library/windows/desktop/ms633591(v=vs.85).aspx>`
+            for information on how to retrieve those messages.
+
+    .. note::
+        This will only affect new processes that aren't launched by services. To
+        apply changes to the path or registry to services, the host must be
+        restarted. The ``salt-minion``, if running as a service, will not see
+        changes to the environment until the system is restarted. Services
+        inherit their environment from ``services.exe`` which does not respond
+        to messaging events. See
+        `MSDN Documentation <https://support.microsoft.com/en-us/help/821761/changes-that-you-make-to-environment-variables-do-not-affect-services>`
+        for more information.
+
+    CLI Example:
+
+    ... code-block:: python
+
+        import salt.utils.win_functions
+        salt.utils.win_functions.broadcast_setting_change('Environment')
+    '''
+    # Listen for messages sent by this would involve working with the
+    # SetWindowLong function. This can be accessed via the win32gui or through
+    # ctypes. You can find examples on how to do this by searching for
+    # `Accessing WGL_WNDPROC` on the internet. Here are some examples of how
+    # this might work:
+    #
+    # # using win32gui
```

Comment:
Why are all of these lines commented out? Do they need to be here at all?

Reply:
These are examples of how to access the broadcast message.  In case someone in the future needs to read the message broadcast by this function. @damon-atkins suggested that this be added,